src/main/java/org/folio/rest/impl/AccountsAPI.java

```diff
@@ -324,54 +323,69 @@ public void putAccountsByAccountId(String accountId, String lang,
-    }
+  @Override
+  public void postAccountsCheckPayByAccountId(String accountId, AccountsCheckRequest request,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
 
-    private void validateAction(String accountId, AccountsCheckRequest request,
-     Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
-     Context vertxContext) {
+    validateAction(accountId, request, okapiHeaders, asyncResultHandler, vertxContext);
+  }
 
-      String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(OKAPI_HEADER_TENANT));
-      PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+  @Override
+  public void postAccountsCheckWaiveByAccountId(String accountId, AccountsCheckRequest request,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
 
-      AccountValidationService validationService = new AccountValidationService(
-        new AccountRepository(pgClient));
-      Double amount = request.getAmount();
+    validateAction(accountId, request, okapiHeaders, asyncResultHandler, vertxContext);
+  }
 
-      validationService.validate(accountId, amount)
-        .onSuccess(result -> {
-          if (result.isValid()) {
-            AccountsCheckResponse response = createBaseAccountCheckResponse(accountId, amount)
-              .withAllowed(true)
-              .withRemainingAmount(((ActionValidationSuccess) result).getRemainingAmount());
-            asyncResultHandler.handle(Future.succeededFuture(
-              PostAccountsCheckPayByAccountIdResponse
-                .respond200WithApplicationJson(response)));
-          } else {
-            AccountsCheckResponse response = createBaseAccountCheckResponse(accountId, amount)
-              .withAllowed(false)
-              .withErrorMessage(((ActionValidationFailure) result).getErrorMessage());
-            asyncResultHandler.handle(Future.succeededFuture(
-              PostAccountsCheckPayByAccountIdResponse
-                .respond422WithApplicationJson(response)));
-          }
-        }).onFailure(e -> Future.succeededFuture(
-        PostAccountsCheckPayByAccountIdResponse.respond500WithTextPlain(e.getMessage())));
-    }
+  private void validateAction(String accountId, AccountsCheckRequest request,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(OKAPI_HEADER_TENANT));
+    PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+
+    FeeFineActionValidationService validationService = new FeeFineActionValidationService(
+      new AccountRepository(pgClient));
+    String rawAmount = request.getAmount();
+    validationService.validate(accountId, rawAmount)
```

Comment:
Blank line?

Reply:
Sorry, I didn't get that. Why the blank line should be here?